galley/pkg/runtime/projections/serviceentry/handler.go

```diff
@@ -261,26 +266,34 @@ func (p *Handler) versionString() string {
 	return fmt.Sprintf("%d", p.version)
 }
 
-func (p *Handler) toMcpResource(svcEntry resource.Entry, endpoints *coreV1.Endpoints) (*mcp.Resource, bool) {
-	fullName := svcEntry.ID.FullName
-	svc := svcEntry.Item.(*coreV1.ServiceSpec)
-	svcMeta := svcEntry.Metadata
-	se := convert.Service(svc, svcMeta, svcEntry.ID.FullName, p.domainSuffix)
+func (p *Handler) newServiceEntry(serviceResource resource.Entry, endpoints *coreV1.Endpoints) *networking.ServiceEntry {
+	se := &networking.ServiceEntry{}
```

Comment:
Similarly a comment here could be useful about why preallocating this is useful.

Reply:
comment no longer applies.